librust/proc_macro/rust/bridge/span.rs

```diff
@@ -0,0 +1,32 @@
+//! Bridge internal span representation and functions
+//!
+//! # Note
+//!
+//! All methods accessing source location in rust are unstable, hence this
+//! implementation with an empty structure.
+
+#[derive(Copy, Clone, Debug)]
+#[repr(C)]
+pub struct Span {}
+
+impl Span {
+    pub fn call_site() -> Self {
+        Span {}
```

Comment:
It is necessary to distinguish between call_site, expansion_site and mixed_site for macro hygiene purposes. In addition for spans originating from the macro input, there is macro hygiene that needs to be recorded too. I'm not sure if gccrs already implements macro hygiene, but if it doesn't, omitting it here for now shouldn't matter.